lib/spack/spack/main.py

```diff
@@ -932,10 +1120,16 @@ def _main(argv=None):
     # main() is tricky to get right, so be careful where you put things.
     #
     # Things in this first part of `main()` should *not* require any
-    # configuration. This doesn't include much -- setting up th parser,
+    # configuration. This doesn't include much -- setting up the parser,
     # restoring some key environment variables, very simple CLI options, etc.
     # ------------------------------------------------------------------------
 
+    # We want the git warnings to ALWAYS be shown before anything else can happen
+    status = perform_git_checks()
+    if status != 0:
+        return status
+    del status
```

Comment:
I don't think we want this (require to have `git` available and potentially wait a long time on each command).